coding-standard/action.yml

```diff
@@ -0,0 +1,138 @@
+        composer require --dev "digitalrevolution/php-codesniffer-baseline:${{ inputs.baseline_version }}"
+
+    - name: Register Coding Standard
+      shell: bash
+      working-directory: standard
+      run: vendor/bin/phpcs --config-set installed_paths ${{ github.workspace }}/standard/vendor/magento/magento-coding-standard,${{ github.workspace }}/standard/vendor/phpcompatibility/php-compatibility
+
+    - name: Get all changed files
+      id: changed-files
+      uses: tj-actions/changed-files@v35
+
+    - name: Checkout - Before Merge
+      shell: bash
+      working-directory: standard
+      id: previous-commit-files
+      run: |
+        if ${{ github.event_name == 'pull_request' }}; then
+          git checkout ${{ github.event.pull_request.base.ref }}
+        else
+          git checkout ${{ github.event.before }}
+        fi
+
+        for file in ${{ steps.changed-files.outputs.all_modified_files }}; do
+          echo $file >> ${{ github.workspace }}/file_lists.txt
+        done
+        cat ${{ github.workspace }}/file_lists.txt
+
+    - name: Filter php files and execute phpcs - Before Merge
+      shell: bash
+      id: phpcs_before
+      working-directory: standard
+      run: |
+        grep -i "\.php$" ${{ github.workspace }}/file_lists.txt > ${{ github.workspace }}/phpcs_files.txt
+        grep -i "\.phtml$" ${{ github.workspace }}/file_lists.txt >> ${{ github.workspace }}/phpcs_files.txt
+
+        echo $(php vendor/bin/phpcs --standard=Magento2 \
+        $([ -n "${{ inputs.severity }}" ] && echo "--severity=${{ inputs.severity }}") \
+        $([ -n "${{ inputs.warning_severity }}" ] && echo "--warning-severity=${{ inputs.warning_severity }}") \
+        $([ -n "${{ inputs.error_severity }}" ] && echo "--error-severity=${{ inputs.error_severity }}") \
+        --report=\\DR\\CodeSnifferBaseline\\Reports\\Baseline --file-list=${{ github.workspace }}/phpcs_files.txt) > ${{ github.workspace }}/phpcs_old.xml
+
+    - name: Checkout - After Merge
+      shell: bash
+      working-directory: standard
+      id: latest-files
+      run: |
+        if ${{ github.event_name == 'pull_request' }}; then
+          git checkout ${{ github.event.pull_request.head.ref }}
+        else
+          git checkout ${{ github.event.after }}
+        fi
+
+    - name: Filter php files and execute phpcs - After Merge
+      shell: bash
+      id: phpcs_after
+      working-directory: standard
+      run: |
+        php vendor/bin/phpcs --standard=Magento2 \
+        $([ -n "${{ inputs.severity }}" ] && echo "--severity=${{ inputs.severity }}") \
+        $([ -n "${{ inputs.warning_severity }}" ] && echo "--warning-severity=${{ inputs.warning_severity }}") \
+        $([ -n "${{ inputs.error_severity }}" ] && echo "--error-severity=${{ inputs.error_severity }}") \
+        --report=\\DR\\CodeSnifferBaseline\\Reports\\Baseline --file-list=${{ github.workspace }}/phpcs_files.txt
```

Comment:
The second call should not generate the baseline? In the second call, the baseline from above should automatically be used. We should not define `--report` here, because we want to use the default report from phpcs.

Reply:
will check and update

Reply:
Have discussed with Vinai, and as per our requirement we need to get baseline report only for changed lines and for that we need to run the baseline 2 times.  1 - Before Merge applied - that will create the Baseline after merging has been done. In the next time, the CodeSnifferBaseline module verify the changes of Before and After merge and remove the older one from the Baseline report. So we require to run command 2 times

Reply:
Not quite. We need to have two steps:

1. Check out old code and generate baseline file.
2. Check out new code and run `phpcs` again. Since there is a baseline file now, all old issues are ignored and only new ones are reported. However, during this call, we want the default `phpcs` output - we do NOT want to create a baseline file again.